### DISCUSSION TITLE
STM32F1xx in Arduino nano size

### DISCUSSION THREAD

**Discussion by Volksolive** [grblHAL/core#8] [General]
Some time ago I have designed a board based on the STM32F103cbt6 which has the same form factor and pin placement has a Arduino nano. The idea was to replace an arduino nano with 32 bits micro controller on these cheap 3 axis GRBL CNC shield widely available. I haven't shared the design on github but if there is people interested here, I would gladly share my design. A little correction to the PCB would be needed first but it is just a quick fix.

**Comment by terjeio** [comment]
You should consider a switch to a F303 instead of the F103? Same footprint but a little different pin assignments? The F103 is at the limits both for RAM and FLASH - and I may "freeze" the driver at some core version in the near future.

**Comment by pagabee** [comment]
Which controllers do you plan to support long term?
STM32F4xx?

**Reply by terjeio** [reply]
The STM32F1xx and MSP430F5529 drivers are likely to be frozen to a specific core version at some point as they have little memory left for new features. They should not be used for new installations.

Which driver is better choose for a particular machine is another matter - that depends on your needs. Board and breakout-board availability should be part of that choice, some will surely fade into oblivion as time passes.
